src/components/Chat/ChatMessage.tsx

```diff
@@ -46,50 +64,91 @@ class ChatMessage extends UIComponent<Extendable<IChatMessageProps>, any> {
     /** Custom styles to be applied for component. */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
+    /** Timestamp of the message. */
+    timestamp: PropTypes.string,
+
     /** Custom variables to be applied for component. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
   }
 
   static handledProps = [
     'as',
+    'author',
     'avatar',
     'children',
     'className',
     'content',
     'mine',
     'styles',
+    'timestamp',
     'variables',
   ]
 
   static defaultProps = {
     as: 'li',
   }
 
-  renderComponent({ ElementType, classes, rest, styles, variables }) {
-    const { avatar, children, content, mine } = this.props
+  renderComponent({
+    ElementType,
+    classes,
+    rest,
+    styles,
+    variables,
+  }: IRenderResultConfig<IChatMessageProps>) {
+    const { as, avatar, children, mine } = this.props
 
     return childrenExist(children) ? (
       <ElementType {...rest} className={cx(classes.root, classes.content)}>
         {children}
       </ElementType>
     ) : (
-      <ElementType {...rest} className={classes.root}>
-        {!mine && this.renderAvatar(avatar, styles, variables)}
-        <div className={classes.content}>{content}</div>
-        {mine && this.renderAvatar(avatar, styles, variables)}
-      </ElementType>
+      <Layout
+        as={as}
+        {...rest}
+        className={classes.root}
+        start={!mine && this.renderAvatar(avatar, styles.avatar, variables)}
+        main={this.renderContent(classes.content)}
+        end={mine && this.renderAvatar(avatar, styles.avatar, variables)}
+      />
     )
   }
 
+  private renderContent = (contentClass: string) => {
+    const { author, content, mine, timestamp } = this.props
+
+    return (
+      <Layout
+        className={contentClass}
+        vertical
+        start={
+          <>
+            {!mine && author && this.renderText(author)}
+            {timestamp && this.renderText(timestamp, true)}
+          </>
+        }
+        main={content}
+      />
+    )
+  }
+
+  private renderText = (content: string, timestamp?: boolean) => (
+    <Text
+      timestamp={timestamp}
```

Comment:
lets use two separate components for that - one for displaying `timestamp`, another one - to display `content`. This is necessary because `Text` component is seen to be a basic one, so it would support either `timestamp` variant, or regular `content` one. More than that, it is highly likely that `timestamp` variant won't be introduced as part of core Stardust